### PR TITLE
[core]Fix partition path error for binary and varbinary partition type

### PR DESCRIPTION
**Pull request by shidayang**
### Purpose
I encounter FileNotFoundException when quering table with varbinary partition.
This is cause by generating file paths directly using the toString of the byte[].

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/utils/RowDataPartitionComputer.java [hunk 1]
```diff
@@ -52,7 +55,19 @@ public LinkedHashMap<String, String> generatePartValues(InternalRow in) {
 
         for (int i = 0; i < partitionFieldGetters.length; i++) {
             Object field = partitionFieldGetters[i].getFieldOrNull(in);
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/utils/RowDataPartitionComputer.java:57]: Can we introduce a StringFormatter to replace FieldGetter?

Comment (shidayang) [reply paimon-core/src/main/java/org/apache/paimon/utils/RowDataPartitionComputer.java:57]: Good idea

File: paimon-common/src/main/java/org/apache/paimon/utils/StringFormatter.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            DataType dataType, Function<T, Object> dataGenerator) {
+        switch (dataType.getTypeRoot()) {
+            case BINARY:
+            case VARBINARY:
+                return row -> {
+                    byte[] bytes = (byte[]) dataGenerator.apply(row);
+                    if (bytes == null) {
+                        return null;
+                    }
+                    return StandardCharsets.ISO_8859_1
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/utils/StringFormatter.java:53]: I'm not sure if this is the standard to string method, is your PR from a real business? Can we simply prohibit partition fields from being bytes type in the SchemeValidation?

Comment (shidayang) [reply paimon-common/src/main/java/org/apache/paimon/utils/StringFormatter.java:53]: This is not from actual business. Not allowing binary to be a partition is a good way to handle it.

**Pull request closed without merging**